### ISSUE TITLE
getting an error from the master branch

### ISSUE THREAD

**Issue by den-run-ai** [shap/shap#124]
Running the census notebook right from the master and the package installed from the master branch too:

https://shap-denfromufa.notebooks.azure.com/nb/notebooks/notebooks/Census%20income%20classification%20with%20Keras.ipynb

https://notebooks.azure.com/denfromufa/libraries/SHAP/html/notebooks/Census%20income%20classification%20with%20Keras.ipynb

```python
  0%|          | 0/50 [00:00<?, ?it/s]
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-8-11a3fe4b051d> in <module>()
----> 1 shap_values50 = explainer.shap_values(X.iloc[280:330,:], nsamples=500)

~/library/shap/explainers/kernel.py in shap_values(self, X, **kwargs)
    198                 if self.keep_index:
    199                     data = convert_to_instance_with_index(data, column_name, index_value[i:i + 1], index_name)
--> 200                 explanations.append(self.explain(data, **kwargs))
    201 
    202             # vector-output

~/library/shap/explainers/kernel.py in explain(self, incoming_instance, **kwargs)
    360             phi_var = np.zeros((len(self.data.groups), self.D))
    361             for d in range(self.D):
--> 362                 vphi, vphi_var = self.solve(self.nsamples / self.max_samples, d)
    363                 phi[self.varyingInds, d] = vphi
    364                 phi_var[self.varyingInds, d] = vphi_var

~/library/shap/explainers/kernel.py in solve(self, fraction_evaluated, dim)
    437 
    438     def solve(self, fraction_evaluated, dim):
--> 439         eyAdj = self.linkfv(self.ey[:, dim]) - self.link.f(self.fnull[dim])
    440         s = np.sum(self.maskMatrix, 1)
    441 

IndexError: invalid index to scalar variable.
```

**Comment by slundberg** [comment]
Thanks for reporting this. I'll see what I can find after I finish with
some un-merged changes.

**Comment by den-run-ai** [comment]
@slundberg i checked all versions of this shap package and the notebooks all error out on the IndexError. Is there any combination of the notebook and shap package that definitely works?

**Comment by den-run-ai** [comment]
@slundberg i also noticed that you don't set github/git tags for correspondinng shap versions released on pypi. So it is hard to troubleshoot the regression.

**Comment by slundberg** [comment]
You are right, I should be making git tags when I push things to make this easier.

I found the error. It comes from calling shap_values() twice on the same explainer when not using vector output. It's a simple fix that I will push once I finish with a Keras DeepExplainer implementation I am in the middle of. For now just rebuilding the explainer object before calling shap_values will fix the issue.

Thanks for pointing this out!

**Comment by slundberg** [comment]
This should be fixed now in 0.19.1
